pkg/test/builder/taskrun.go

```diff
@@ -14,3 +15,23 @@ func TaskRunCompletionTime(ct time.Time) tb.TaskRunStatusOp {
 		s.CompletionTime = &metav1.Time{Time: ct}
 	}
 }
+
+// StepState adds a StepState to the TaskRunStatus.
+func StepState(ops ...tb.StepStateOp) tb.TaskRunStatusOp {
+	return func(s *v1alpha1.TaskRunStatus) {
+		state := &v1alpha1.StepState{}
+		for _, op := range ops {
+			op(state)
+			s.Steps = append(s.Steps, *state)
+		}
+	}
+}
+
+func StateTerminated(name string, exitcode int) tb.StepStateOp {
+	return func(s *v1alpha1.StepState) {
+		s.Name = name
```

Comment:
Shouldn't this be split into 2 operation : `StepStateTerminated` and `StepStateName` or something ?

Reply:
That's a good suggestion, let's see if we could.

Reply:
good catch :+1:  , resolved it  :sweat_smile: